src/Features/Core/Portable/Workspace/BackgroundParser.cs

```diff
@@ -163,17 +165,6 @@ public void Parse(Document document)
                 {
                     CancelParse(document.Id);
 
-                    if (SolutionCrawlerOptions.GetBackgroundAnalysisScope(document.Project) == BackgroundAnalysisScope.ActiveFile &&
-                        _documentTrackingService?.TryGetActiveDocument() != document.Id)
```

Comment:
I removed this "extra" early return. We still return early from the scheduled task, but the code is cleaner.